sdk/include/xdk/amd64/ke.h

```diff
@@ -320,11 +329,20 @@ KeGetPcr(VOID)
     return (PKPCR)__readgsqword(FIELD_OFFSET(KPCR, Self));
 }
 
+#if (NTDDI_VERSION >= NTDDI_WIN7)
+_CRT_DEPRECATE_TEXT("KeGetCurrentProcessorNumber is deprecated. Use KeGetCurrentProcessorNumberEx or KeGetCurrentProcessorIndex instead.")
+#endif
 FORCEINLINE
 ULONG
 KeGetCurrentProcessorNumber(VOID)
 {
-    return (ULONG)__readgsword(0x184);
+#if (NTDDI_VERSION >= NTDDI_VISTA)
+    return __readgsword(0x184);
+#elif (NTDDI_VERSION >= NTDDI_WS2003)
+    return __readgsbyte(0x184);
+#else
+    return __readgsbyte(FIELD_OFFSET(KPCR, Number));
```

Comment:
AMD64 versions pre-Win2003 don't exist, so this seems pointless

Reply:
@ThFabba: Windows XP x64 does exist (I have tried it) but it is probably more similar to WS2003 then 32-bit XP.